### PR TITLE
Adding erc4906 protocol

### PR DESCRIPTION
**Pull request by iainnash**
https://eips.ethereum.org/EIPS/eip-4906

### CODE REVIEW

File: src/ERC721Drop.sol [hunk 1]
```diff
@@ -1164,6 +1201,8 @@ contract ERC721Drop is
             super.supportsInterface(interfaceId) ||
             type(IOwnable).interfaceId == interfaceId ||
             type(IERC2981Upgradeable).interfaceId == interfaceId ||
+            // Because the EIP-4906 spec is event-based a numerically relevant interfaceId is used.
+            bytes4(0x49064906) == interfaceId ||
```
Comment (tbtstl) [inline src/ERC721Drop.sol:1205]: til

File: test/ERC721Drop.t.sol [hunk 2]
```diff
@@ -360,30 +359,55 @@ contract ERC721DropTest is Test {
...
         );
-        calls[2] = abi.encodeWithSelector(
-            IERC721Drop.saleDetails.selector
-        );
+        calls[2] = abi.encodeWithSelector(IERC721Drop.saleDetails.selector);
         bytes[] memory results = zoraNFTBase.multicall(calls);
 
-        (bool saleActive, bool presaleActive, uint256 publicSalePrice, , , , , , , ,) = abi.decode(results[2], (bool, bool, uint256, uint64, uint64, uint64, uint64, bytes32, uint256, uint256, uint256));
+        (
+            bool saleActive,
```
Comment (tbtstl) [inline test/ERC721Drop.t.sol:373]: nit: can we change our linter to keep these single line?

**Review by tbtstl — APPROVED** [review]

File: test/ERC721Drop.t.sol (hunk 2 above)
Comment (iainnash) [reply test/ERC721Drop.t.sol:373]: Oof this is annoying. Can check.

Comment (iainnash) [reply test/ERC721Drop.t.sol:373]: Seems to be more trouble than it's worth – i'll revisit this soon.

**Comment by 0xTranqui** [comment]
just looked thru this PR bc adding it to AssemblyPress. why does this need to be added to the base drop impl and cant just live in the metadata renderer? also am I wrong in believing this is still incomplete bc the renderers we are using in prod (edition + drop) dont emit this event yet on update?

**Comment by iainnash** [comment]
the event needs to come from the token contract.

**Comment by 0xTranqui** [comment]
got it -- so does that mean that the way youve implemented in ERC721Drop, the only way to trigger the event is to make some type of arbitrary call ? are you planning on linking the EditionMetadataRenderer update function back to ERC721Drop to trigger that event to automate thigns?

**Pull request merged**